src/routes/v1/secret_test.go

```diff
@@ -11,76 +12,376 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestCreateSecret(t *testing.T) {
-	secretRequest := types.CreateSecretRequest{
-		Type:       "opaque",
-		SecretName: "new-secret",
-		Data:       []types.KeyValue{{Key: "key1", Value: "ZmFrZQ=="}},
-	}
-	body, _ := json.Marshal(secretRequest)
-	request, _ := http.NewRequest("POST", "/v1/namespaces/default/secrets/", bytes.NewBuffer(body))
-	request.Header.Set("Content-Type", "application/json")
-
-	writer := httptest.NewRecorder()
-	router.ServeHTTP(writer, request)
-
-	assert.Equal(t, http.StatusOK, writer.Code)
-	var response types.CreateSecretResponse
-	err := json.Unmarshal(writer.Body.Bytes(), &response)
-	assert.NoError(t, err)
-	assert.Equal(t, "new-secret", response.SecretName)
-	assert.Equal(t, "default", response.NamespaceName)
+func TestGetSecret(t *testing.T) {
+	secretNamespace := "test-namespace-secret"
+	secretName := "test-secret"
```

Comment:
The setup in different tests is done in a different way

https://github.com/dana-team/platform-backend/blob/1989d979499464dac34fd2dbb08b0ea01c502f3a/src/routes/v1/capp_test.go#L26-L32

Reply:
but every test need different setup very impotent concept in unit testing is isolation.